fix: tmp folder not deleting

BD103 opened the pull request:
- Changed errors to be plain text instead of JSON so its more readable
- Fixed a bug where temporary folders would not get deleted if an exception occurred

Currently trying to figure out how Node importing works, I'll be back if I can get PrismJS to cooperate.

Commit:
Make errors readable

Commit:
Fixed fetchFile temp folder buildup

rayhanadev commented on utils/file.js:
```diff
@@ -92,6 +93,7 @@ export async function fetchFiles(urlPath = '/') {
 
 	const output = [];
 
+  // Fix necessary?
```
Elaborate pls :).

BD103 replied:
I wasn't sure if that was where the folder wasn't deleting, or if it was the other one. I don't see any issues, but it may raise a problem later on.

rayhanadev replied:
Gotcha, I'll take a look as well.

rayhanadev commented on utils/file.js:
```diff
@@ -53,10 +53,11 @@ export async function fetchFile(urlPath = '/', fileName, raw = '0') {
 		if (raw === '1') return zip.readAsText(entry);
 		return JSON.stringify([zip.readAsText(entry)]);
 	} catch (err) {
+    rimraf(tmpPath, () => {
+      console.log("Deleted temp folder")
+    });
```
Ooh good catch, didn't think of that :D.

rayhanadev reviewed (changes requested):
✨ Awesome PR dude! This gets a lot of kewl things done :). ✨ 

Just a few heads up that I would like addressed b4 merging.
- We use eslint and prettier. If you would please run each of these (`npm run lint:fix` and `npm run format:fix`) before a PR on any ReplAPI.it project that would be spectacular and helps keep our code clean.
- We follow the [Conventional Commits](https://www.conventionalcommits.org/) convention. I assume you haven't used it before so I'll let it slide this time, but in the future be sure to use it.

Otherwise it looks good! Just run linting and formatting and I'll merge it. I made a few comments on the diff that you might check out as well.

rayhanadev commented on index.js:
```diff
@@ -35,10 +35,7 @@ app.get('/file/:username/:slug', async (req, res) => {
-				error:
-					'An Error Occured. This is likely because the Repl does not exist or it does not have files.',
-			})
+			'An Error Occured. This is likely because the Repl does not exist or it does not have files.'
```
Any specific reason you feel errors should be plain text? I'm willing to switch over on the other API as well if there's a reason.

BD103 replied:
It's a more user-friendly format, instead of JSON. This change depends on whether you think this API is for robots or for humans. If for robots (programs and UIs), then JSON is easier so the robot could detect errors. If it is meant to be used by a human, then this plain text format is more readable. (Or we could use an in-between based on user-agent.)

rayhanadev replied:
The API right now veers toward robots (aka the ReplAPI.it Package method) but I really like the user-agent idea and I think it works. Never heard of it before so I think you might want to tackle it, but it sounds like a plan to me :).

BD103 replied:
Ok. I'll make another branch for the user-agent feature. For now I'll revert back to JSON format.

RayhanADev commented:
@BD103 I've invited you to the team on Replit, so it might be easier for you to code on the main Repl after this PR.

Commit:
fix: Revert change from 50f80cd5d

Commit:
fix: lint and fix

Commit:
fix: Catch error on fetchFiles()

BD103 commented:
Ok, I fixed it as you requested.

- Errors now return JSON, though I can see an issue with this. We should probably use HTTP status codes instead, because what if you request a JSON file that has an item named `error`?
- Everything is linted and formatted
- I also added a try / except block to `fetchFiles()`. I had to make the zip variable using `let`, instead of `const`, but it should be fine.

Thanks for the invite!

rayhanadev reviewed (approved):
Awesome! This works, and thanks for linting btw ;D.
